wagtail_transfer/static_src/components/ContentImportForm/index.js

```diff
@@ -25,28 +25,7 @@ function SourceSelectorWidget({ sources, selectedSource, onChange }) {
   );
 }
 
-function SubmitButton({ apiBaseUrl, sourcePage, onClick, disabled }) {
-  const [numPages, setNumPages] = React.useState(null);
-
-  React.useEffect(() => {
-    // Fetch descendant count whenever sourcePage is changed
-    if (numPages !== null) {
-      setNumPages(null);
-    }
-
-    if (sourcePage) {
-      const fetchNumPages = async () => {
-        const api = new PagesAPI(apiBaseUrl);
-        const page = await api.getPage(sourcePage.id, {
-          fields: 'descendants'
-        });
-
-        setNumPages(page.meta.descendants.count + 1);
-      };
-
-      fetchNumPages();
-    }
-  }, [sourcePage]);
+function SubmitButton({ apiBaseUrl, sourcePage, onClick, disabled, numPages }) {
```

Comment:
Definitely nicer with the state moved up into the form component, thanks! We can get rid of the `apiBaseUrl` and `sourcePage` parameters now.

Reply:
No problem! I've removed them now, thanks.